src/lib.rs

```diff
@@ -155,15 +155,15 @@ impl<'t> DoRead<'t> {
         let size = unsafe {
             // TODO: This can be replaced by std::mem::MaybeUninit::first_ptr_mut() once
             // it is stabilized.
-            let data = &mut v.bytes_mut()[..self.reserve];
+            let data = &mut v.chunk_mut()[..self.reserve];
             // We first have to initialize the data or otherwise casting to a byte slice
             // below is UB. See also code of std::io::copy(), tokio::AsyncRead::poll_read_buf()
             // and others.
             //
             // Read::read() might read uninitialized data otherwise, and generally creating
             // references to uninitialized data is UB.
-            for x in data.iter_mut() {
-                *x.as_mut_ptr() = 0;
+            for i in 0..data.len() {
+                data.write_byte(i, 0);
```

Comment:
Or maybe instead of that we can do something with `bytes::buf::UninitSlice` here without having to initialize everything first?